File: baseline/src/tuples.rs
use std::collections::HashSet;
use std::collections::HashMap;
pub fn use_tuple_for_swap(mut a: i64, mut b: i64) -> i64 {
    if b > a {
        let tmp = (b, a);
        a = tmp.0;
        b = tmp.1;
    }
    return a - b;
}

pub fn return_tuple(a: i64, b: i64) -> (i64, i64) {
    return (a, b);
}

pub fn use_returned_tuple(a: i64, b: i64) -> i64 {
    let (c, d) = return_tuple(a, b);
    return use_tuple_for_swap(c, d);
}

pub fn tuple_in_args(mut a: (i64, i64)) -> i64 {
    if a.1 > a.0 {
        a = (a.1, a.0);
    }
    return a.0 - a.1;
}

File: tests/tuples.py
from typing import Tuple

def use_tuple_for_swap(a: int, b: int) -> int:
    if b > a:
        a, b = b, a
    return a - b

def return_tuple(a: int, b: int) -> Tuple[int, int]:
    return (a, b)

def use_returned_tuple(a: int, b: int) -> int:
    c, d = return_tuple(a, b)
    return use_tuple_for_swap(c, d)

def tuple_in_args(a: Tuple[int, int]) -> int:
    if a[1] > a[0]:
        a = a[1], a[0]
    return a[0] - a[1]
